sdk/storage/azure-storage-files-datalake/src/datalake_file_system_client.cpp

```diff
@@ -160,48 +160,49 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
     m_pipeline = std::make_shared<Azure::Core::Http::HttpPipeline>(policies);
   }
 
-  PathClient FileSystemClient::GetPathClient(const std::string& path) const
+  DataLakePathClient DataLakeFileSystemClient::GetPathClient(const std::string& path) const
   {
     auto builder = m_dfsUri;
     builder.AppendPath(Storage::Details::UrlEncodePath(path));
-    return PathClient(builder, m_blobContainerClient.GetBlobClient(path), m_pipeline);
+    return DataLakePathClient(builder, m_blobContainerClient.GetBlobClient(path), m_pipeline);
   }
 
-  FileClient FileSystemClient::GetFileClient(const std::string& path) const
+  DataLakeFileClient DataLakeFileSystemClient::GetFileClient(const std::string& path) const
   {
 
     auto builder = m_dfsUri;
     builder.AppendPath(Storage::Details::UrlEncodePath(path));
     auto blobClient = m_blobContainerClient.GetBlobClient(path);
     auto blockBlobClient = blobClient.AsBlockBlobClient();
-    return FileClient(
+    return DataLakeFileClient(
         std::move(builder), std::move(blobClient), std::move(blockBlobClient), m_pipeline);
   }
 
-  DirectoryClient FileSystemClient::GetDirectoryClient(const std::string& path) const
+  DataLakeDirectoryClient DataLakeFileSystemClient::GetDirectoryClient(
+      const std::string& path) const
```

Comment:
Is `path` misleading here? Because I remember we only allow changing one directory level for each call.

Reply:
I don't remember we extended this also to the DataLake service.

Reply:
I don't know, whatever

Reply:
I watched the [recording](https://github.com/Azure/azure-sdk/issues/1983#issuecomment-742659811) again, at 1:44:02 
you asked "It's also true for the DataLake service, right Ted?". 
And then Ted answered "Correct, we only take names, never path segments.", 
then you said "OK".